endpoints/openrtb2/auction.go

```diff
@@ -1340,21 +1340,38 @@ func (deps *endpointDeps) processStoredRequests(ctx context.Context, requestJson
 
 	// Apply the Stored BidRequest, if it exists
 	resolvedRequest := requestJson
-	if hasStoredBidRequest {
-		resolvedRequest, err = jsonpatch.MergePatch(storedRequests[storedBidRequestId], requestJson)
+
+	// If GenerateRequestId is true, we'll generate a uuid for the bid request
+	if deps.cfg.GenerateRequestId && !hasStoredBidRequest {
+		uuidPatch, err := generateUuidForBidRequest()
 		if err != nil {
-			hasErr, Err := getJsonSyntaxError(requestJson)
-			if hasErr {
-				err = fmt.Errorf("Invalid JSON in Incoming Request: %s", Err)
-			} else {
-				hasErr, Err = getJsonSyntaxError(storedRequests[storedBidRequestId])
-				if hasErr {
-					err = fmt.Errorf("Invalid JSON in Stored Request with ID %s: %s", storedBidRequestId, Err)
-					err = fmt.Errorf("ext.prebid.storedrequest.id refers to Stored Request %s which contains Invalid JSON: %s", storedBidRequestId, Err)
-				}
-			}
 			return nil, nil, []error{err}
 		}
+		resolvedRequest, err = jsonpatch.MergePatch(requestJson, uuidPatch)
+		if err != nil {
+			return nil, nil, []error{err}
+		}
+	} else if deps.cfg.GenerateRequestId && hasStoredBidRequest {
+		uuidPatch, err := generateUuidForBidRequest()
+		if err != nil {
+			return nil, nil, []error{err}
+		}
+		storedReqForPatch, err := jsonpatch.MergePatch(storedRequests[storedBidRequestId], uuidPatch)
+		if err != nil {
+			requestErrorJson, errL := storedRequestErrorChecker(err, requestJson, storedRequests, storedBidRequestId)
+			return requestErrorJson, nil, errL
+		}
+		resolvedRequest, err = jsonpatch.MergePatch(requestJson, storedReqForPatch)
+		if err != nil {
+			requestErrorJson, errL := storedRequestErrorChecker(err, requestJson, storedRequests, storedBidRequestId)
+			return requestErrorJson, nil, errL
+		}
+	} else if !deps.cfg.GenerateRequestId && hasStoredBidRequest {
```

Comment:
If you reach this line, and `hasStoredBidRequest` is true, then you know `!deps.cfg.GenerateRequestId` also has to be true, you don't need to test both.

Another potential logic flow is:
```suggestion
        storedReqForPatch = ''
        if hasStoredBidRequest {
          storedReqForPatch = storedRequests[storedBidRequestId]
        }
	if deps.cfg.GenerateRequestId {
	  uuidPatch, err := generateUuidForBidRequest()
	  if hasStoredBidRequest {
	    storedReqForPatch, err := jsonpatch.MergePatch(storedRequests[storedBidRequestId], uuidPatch)
	  } else {
	    storedReqForPatch := uuidPatch
	    hasStoredBidRequest = true
	  }
	}
	
	if hasStoredBidRequest {
	  resolvedRequest, err = jsonpatch.MergePatch(requestJson, storedReqForPatch)
	}
```
with the error handling and other supporting code included.